File: ShopAppSpring/src/main/java/com/duchung/shopappspring/dtos/RawProductDTO.java
package com.duchung.shopappspring.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class RawProductDTO {

    private String name;

    private double price;

    private Long quantity;

    private String note;

    @JsonProperty("provider_id")
    private Long providerId;
}

File: shopapp-angular/src/app/manager/dtos/raw.product.dto.ts
export class RawProductDTO {
  name: string;
  price: number;
  quantity: number;
  note: string;
  provider_id: number;

  constructor(data: any) {
    this.name = data.name;
    this.price = data.price;
    this.quantity = data.quantity;
    this.note = data.note;
    this.provider_id = data.provider_id;
  }
}
